### DISCUSSION TITLE
Updating DNSValidator regex

### DISCUSSION THREAD

**Discussion by fmlshai** [netbox-community/netbox#8795] [Q&A]
Hi,

like in this commit https://github.com/netbox-community/netbox/commit/95edec54485529f69f9df526ebbc69772b36bc6c
I want to update the DNSValidator regex and create a pull request. A * needs to be added, to allow wildcard domains.
Do I also have to create a migration file like in the mentioned commit? I have zero experience with that.

**Comment by candlerb** [comment]
I don't think a pull request that does that will be accepted.  That field is supposed to record the primary DNS name associated with this IP address (i.e. that name resolves to this address, and that address resolves to this name), and you can't use a wildcard for that.

I believe there are plugins available to Netbox for doing more general DNS management, which I expect would let you create arbitrary DNS resource records.  There's netbox-dns: I've not tried it myself, but it was listed in the [2022 community survey](https://github.com/netbox-community/netbox/discussions/8768).

Migrations are used when you're changing the database schema. In the example you showed there, it was to add a column to a table.

Alternatively, you can add a [Custom Field](https://netbox.readthedocs.io/en/stable/customization/custom-fields/) to the IP Address object, and use whatever regex you like to validate it.

**Reply by fmlshai** [reply]
Well, wildcard records are valid, even though the implementation may vary in different DNS servers. See https://en.wikipedia.org/wiki/Wildcard_DNS_record for example, and the RFCs mentioned there. So I think the DNSValidtor script should accept this.
Yes, I could use a custom field, or the description field. But the field for dns names is there, and it should just accept everything that's allowed for dns records.
I'm going to create the pull request for this. Thank you.

**Reply by kkthxbye-code** [reply]
@fmlshai - Please remember to read the contributor guidelines first. A pull request will only be accepted if there is an accepted issue first.
